src/libLookup/Lookup.cpp

```diff
@@ -2695,8 +2641,33 @@ void Lookup::SendTxnPacketToNodes(uint32_t nShard)
     {
         vector<unsigned char> msg
             = {MessageType::NODE, NodeInstructionType::FORWARDTXNBLOCK};
-        if (!CreateTxnPacket(msg, i, MessageOffset::BODY, mp))
+        bool result = false;
+
+        {
+            lock_guard<mutex> g(m_txnShardMapMutex);
+
+            unsigned int size_dummy
+                = (mp.find(i) != mp.end()) ? mp.at(i).size() : 0;
+            size_dummy = size_dummy / Transaction::GetMinSerializedSize();
+
+            if (size_dummy > 0)
+            {
+                result = Messenger::SetNodeForwardTxnBlock(
+                    msg, MessageOffset::BODY, m_mediator.m_currentEpochNum, i,
+                    m_txnShardMap[i], mp.at(i));
+            }
+            else
+            {
+                result = Messenger::SetNodeForwardTxnBlock(
+                    msg, MessageOffset::BODY, m_mediator.m_currentEpochNum, i,
+                    m_txnShardMap[i], {});
```

Comment:
A log here to log how many txns generated for each shard.

Reply:
Ok, added logs directly inside the Set and Get functions